update IngredientsAnalysisTags

stephanegigandet opened the pull request:
This fixes #130 reported by @blazern . I added the missing tags, and also renamed others so that they match what is returned by the API (some were in fact incorrect). Sio it is a breaking change.

monsieurtanuki commented on lib/model/IngredientsAnalysisTags.dart:
```diff
@@ -1,40 +1,70 @@
     }
 
-    veganStatus ??= VeganStatus.MAYBE;
+    veganStatus ??= VeganStatus.VEGAN_STATUS_UNKNOWN;
```
I would handle the `null` case in `_getVeganStatus`.

stephanegigandet replied:
In fact I kept the code as it was, but we probably should keep null as a possible value, in the case where the API does not send back anything (which should not happen).

monsieurtanuki commented on lib/model/IngredientsAnalysisTags.dart:
```diff
@@ -57,42 +87,51 @@ class IngredientsAnalysisTags {
         break;
+      case VeganStatus.VEGAN_STATUS_UNKNOWN:
+        result.add('en:vegan-status-unknown');
+        break;
```
Perhaps we should say `UNKNOWN` if we don't know. But I'm not sure if it's strictly the same in OFF "not to know" and "know that we don't know".

monsieurtanuki reviewed (commented):
Hi @stephanegigandet!
I'm not 100% ok with some of your choices, but I wouldn't stop the merge of this PR.
Please have a look at my comments.

stephanegigandet commented on lib/model/IngredientsAnalysisTags.dart:
```diff
@@ -1,40 +1,70 @@
+  NON_VEGAN,
+  MAYBE_VEGAN,
+  VEGAN_STATUS_UNKNOWN,
+}
```
The reason I renamed the fields is that they can match the names from the API. If the fields in openfoodfacts-dart have different names than the ones used in the API, it makes things quite difficult to document, debug etc.

So I would rather make a breaking change now (there are probably very few people, if any, that use those fields).

peterwvj replied:
For what it's worth, I prefer to have the names in the SDK closely resemble those of the API so I think the change is fine.

monsieurtanuki replied:
Ok then!

peterwvj reviewed (approved):
Looks good, thanks!

monsieurtanuki reviewed (approved):
That's more or less what I had in mind, good!
Additional remarks:
* would it make sense to make the `status` fields `final`?
* I would have used private lists in order to avoid typos and reuse the same values for from/to String/status conversion
```dart
const static List<String> _VEGAN_TAGS = [
  'en:vegan',
  'en:non-vegan',
  'en:maybe-vegan',
  'en:vegan-status-unknown',
];

const static List<VeganStatus> _VEGAN_STATUSES = [
  VeganStatus.VEGAN,
  VeganStatus.NON_VEGAN,
  VeganStatus.MAYBE_VEGAN,
  VeganStatus.VEGAN_STATUS_UNKNOWN,
];
```

stephanegigandet commented:
> * would it make sense to make the `status` fields `final`?

Well I guess that in theory, they could be updated by an app, which would then be able to generate the corresponding tags. Otherwise the IngredientsAnalysisTags.toJson() function would never be used. 

> * I would have used private lists in order to avoid typos and reuse the same values for from/to String/status conversion

OK. I changed the code to use the lists you suggested, and added a test to make sure the code for the from/to conversion works.
